src/API/Middlewares/HttpExceptionsMiddleware.ts

```diff
@@ -14,10 +14,11 @@ export const HttpExceptionsMiddleware: Koa.Middleware = async (context: any, nex
       context.status = 404
     } else {
       if (context.logger)
-        context.logger.error({ exception }, 'Error Caught at Middleware - Will Return Internal Server Error')
+        context.logger.error({ exception }, 'Error Caught at Middleware')
 
-      context.body = 'Internal Server Error'
-      context.status = 503
+      context.body = exception.message || 'Internal Server Error'
```

Comment:
You don't want to leak error details to the world by default. We need to either keep it hardcoded to 'Internal Server Error' or add some logic and configuration options and leak details in testing/qa but not in prod.